cmd/juju/user/login.go

```diff
@@ -513,16 +513,16 @@ Run "juju logout" first before attempting to log in as a different user.`,
 
 	if c.username == "" {
 		// No username specified, so try external-user login.
-		conn, err := safeDial(&jujuclient.AccountDetails{})
+		accountDetails = &jujuclient.AccountDetails{}
+		conn, err := safeDial(accountDetails)
```

Comment:
In your comment:
"Because of a similar reason to 1. we need to ensure that during login, we don't lose the session token, done with the small change of variable in cmd/juju/user/login.go."

If this is what your comment refers to, could you explain why keeping the reference to accountDetails prevents us losing the token here? It looks like it shouldn't make a difference here?

Reply:
During local the accountDetails.SessionToken field is updated with a session token. We need to keep the changes to this struct so that when we return it, it can be saved with all the data that it was enriched with.